### PR TITLE
feat: flutter SDK updates

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

This PR includes the following updates

- bump dependency versions
- update example code to demonstrate new organization feature
- add more comments in SDK, and provide SDK docs link in both pubspec file and readme.

## Checklist

- [ ] ~`.changeset`~
- [x] ~unit tests~
- [ ] ~integration tests~
- [x] necessary TSDoc comments

### CODE REVIEW

File: lib/logto_client.dart [hunk 1]
```diff
@@ -55,22 +71,26 @@ class LogtoClient {
     return token?.serialization;
   }
 
+  // Get the idToken claims.
```
Comment (gao-sun) [inline lib/logto_client.dart:74]: no need for this since the function name itself do the same job?

File: lib/logto_client.dart [hunk 2]
```diff
@@ -11,13 +10,30 @@ import '/src/modules/pkce.dart';
...
+/**
+ * LogtoClient
+ * 
+ * The main class for the Logto SDK.
+ * It provides all the user authentication and authorization methods.
+ * 
+ * @param config: LogtoConfig - the basic configuration object for the Logto SDK.
+ * @param storageProvider: LogtoStorageStrategy (optional) - default is [InMemoryTokenStorage] used for storing tokens.
+ * @param httpClient: http.Client (optional) - custom [http.Client] to be used for making http requests.
+ *
```
Comment (gao-sun) [inline lib/logto_client.dart:27]: just double check this is the standard dart comment block and can be read by IDEs

File: lib/logto_client.dart [hunk 3]
```diff
@@ -55,22 +71,26 @@ class LogtoClient {
     return token?.serialization;
   }
 
+  // Get the idToken claims.
   Future<OpenIdClaims?> get idTokenClaims async {
     final token = await _tokenStorage.idToken;
     return token?.claims;
   }
 
+  // Get the OIDC provider configuration.
```
Comment (gao-sun) [inline lib/logto_client.dart:80]: ditto

File: lib/logto_client.dart [hunk 4]
```diff
@@ -11,13 +10,30 @@ import '/src/modules/pkce.dart';
 import '/src/modules/token_storage.dart';
 import '/src/utilities/utils.dart' as utils;
 import 'logto_core.dart' as logto_core;
```
Comment (gao-sun) [inline lib/logto_client.dart:12]: double check the underscore style is a dart convention

**Review by gao-sun — APPROVED** [review]
nothing blocking

File: lib/logto_client.dart (hunk 4 above)
Comment (simeng-li) [reply lib/logto_client.dart:12]: yes, we have lint rule applied

File: lib/logto_client.dart (hunk 1 above)
Comment (simeng-li) [reply lib/logto_client.dart:74]: cool removed

File: lib/logto_client.dart (hunk 2 above)
Comment (simeng-li) [reply lib/logto_client.dart:27]: yep multi-line comments

**Pull request merged**